components/List/tests/unit/MDCSelfSizingStereoCellTests.m

```diff
@@ -0,0 +1,105 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialList.h"
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTypography.h"
+#import "MaterialTypographyScheme.h"
+
+@interface MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow : UIWindow
+
+/** Used to override the value of @c preferredContentSizeCategory. */
+@property(nonatomic, copy) UIContentSizeCategory contentSizeCategoryOverride;
+
+@end
+
+@implementation MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = UIContentSizeCategoryLarge;
+  }
+  return self;
+}
+
+- (instancetype)initWithContentSizeCategoryOverride:
+    (UIContentSizeCategory)contentSizeCategoryOverride {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = contentSizeCategoryOverride;
+  }
+  return self;
+}
+
+- (UITraitCollection *)traitCollection {
+  if (@available(iOS 10.0, *)) {
+    UITraitCollection *traitCollection = [UITraitCollection
+        traitCollectionWithPreferredContentSizeCategory:self.contentSizeCategoryOverride];
+    return traitCollection;
+  }
+  return [super traitCollection];
+}
+
+@end
+
+@interface MDCSelfSizingStereoCellTests : XCTestCase
+
+@end
+
+@implementation MDCSelfSizingStereoCellTests
+
+- (void)testSelfSizingStereoCellDynamicTypeBehavior {
+  // Given
+  MDCTypographyScheme *typographyScheme = [[MDCTypographyScheme alloc] init];
+  MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
+  UIFont *titleFont = typographyScheme.subtitle1;
+  // TODO: Investigate what can be done about the line below failing when using .caption instead of
+  // .button. See
+  // https://github.com/material-components/material-components-ios/issues/7424 for
+  // context.
+  UIFont *detailFont = typographyScheme.button;
+  cell.titleLabel.font = titleFont;
+  cell.detailLabel.font = detailFont;
+  cell.mdc_adjustsFontForContentSizeCategory = YES;
+  cell.titleLabel.text = @"Title";
+  cell.detailLabel.text = @"Detail";
+  CGFloat defaultTitleSize = cell.titleLabel.font.pointSize;
+  CGFloat defaultDetailSize = cell.detailLabel.font.pointSize;
+
+  // When
+  MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+      *extraExtraLargeContainer =
+          [[MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow alloc]
+              initWithContentSizeCategoryOverride:UIContentSizeCategoryExtraLarge];
+  [extraExtraLargeContainer addSubview:cell];
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+  CGFloat fontTitleSizeForExtraExtraLargeSizeCategory = cell.titleLabel.font.pointSize;
+  CGFloat fontDetailSizeForExtraExtraLargeSizeCategory = cell.detailLabel.font.pointSize;
```

Comment:
Assigning these two values to variables makes the tests below a bit harder to follow because it requires reading both the assertion plus this assignment. Please just directly use the `cell.*.font.pointSize` properties in the assertions in order to keep what's being directly tested next to what's expected.

Reply:
Addressed in 06ed43c.